File: app/src/main/java/com/dublikunt/nclient/utility/IntentUtility.java
package com.dublikunt.nclient.utility;

import android.app.Activity;
import android.content.Intent;

import androidx.annotation.NonNull;

public class IntentUtility extends Intent {
    public static String PACKAGE_NAME;

    public static void startAnotherActivity(@NonNull Activity activity, Intent intent) {
        activity.runOnUiThread(() -> activity.startActivity(intent));
    }

    public static void endActivity(@NonNull Activity activity) {
        activity.runOnUiThread(activity::finish);
    }
}

File: app/src/main/kotlin/com/dublikunt/nclient/utility/IntentUtility.kt
package com.dublikunt.nclient.utility

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity

object IntentUtility : Intent() {

    fun startAnotherActivity(activity: AppCompatActivity, intent: Intent?) {
        activity.runOnUiThread { activity.startActivity(intent) }
    }

    fun endActivity(activity: AppCompatActivity) {
        activity.runOnUiThread { activity.finish() }
    }
}
